### ISSUE TITLE
Dual division rules

### ISSUE THREAD

**Issue by safiire** [JuliaDiff/DualNumbers.jl#41]
Hi, nice to see a dual number library in Julia.  I have some issues for your consideration.

```
Dual(0, 2) / Dual(0, 6)
NaN + NaN*ɛ
```

When both terms are purely dual, and the denominator is not 0 + 0ε causing a division by zero, I believe this is the correct way to perform division:

Solving for `x, y` in:

```
(0 + bε) / (0 + dε) = (x + yε)
(0 + bε) = (x + yε) * (0 + dε)
(0 + bε) = x * 0 + xdε + yε * 0 + ydε^2
(0 + bε) = xdε
x = bε/dε
x = b/d
y = anything, it is annihilated. 
```

Because of this, in my implementation of Dual numbers dividing by two purely dual numbers results in a real, and I set the dual part to 0ε, (but it could be anything).

By doing this I am able to perform a division, where the result multiplied by the denominator gives you back the numerator, and result multiplied by the numerator gives you back the denominator, keeping the inverse relationship between multiplication and division consistent.

```
(0 + 2ε) /  (0 + 6ε) = (1/3 + 0ε)
(1/3 + 0ε) * (0 + 6ε) = (0 + 2ε)
(1/3 + 0ε) * (0 + 2ε) = (0 + 6ε)
```

I realize it makes no sense for an operation on 2 infinitesimal numbers to result in a real, and the selection of 0 is arbitrary since it has an infinite amount of solutions, but when writing my implementation I searched around on math exchange etc, and found this method as accepted.  You can also think of this as the ε cancelling out during the division, resulting in a real.

What do you think of this?

I have a few more suggestions, I will make separate issues for.  Thanks :)

Here is my implementation, albeit in C++

```
  //// Division
  Dual<T> operator/=(const Dual<T> &other){
    if(_real == 0 && other._real == 0 && other._dual != 0){
      _real = _dual / other._dual;
      _dual = 0;
    }else{
      //  Otherwise we divide the same way as a regular complex number
      //  d * d.conjugate() is always Real(d)^2
      (*this) *= other.conjugate();
      auto reciprocal = 1 / (other._real * other._real);
      _real *= reciprocal;
      _dual *= reciprocal;
    }
    return *this;
  }

  Dual<T> operator/(const Dual<T> &other) const {
    auto copy = *this;
    return copy /= other;
  }
```

**Comment by KristofferC** [comment]
> I realize it makes no sense for an operation on 2 infinitesimal numbers to result in a real, 

I agree with this :) 

From what I have seen division between Duals is typically only defined when the real part of the denominator is non zero.

**Comment by goretkin** [comment]
I understand the temptation of seeing `ε/ε` and calling it `1`, however it's not valid. It would imply that `1/ε` exists, which means that there is dual number `x` such that  `xε=1`. `x` can't be purely real, because you'd end up with a number with an epsilon part. `x`can't be purely epsilon, because then you just end up with zero. And combination in between is a combination of zero and epsilon. You can't make `1` out of that.

In short, `ε` does not have an inverse.

(Note also, that you can start with a false statement and get to a true one by multiplying both sides by ε. For example `3ε=4ε`, so you must take care with any argument that multiples or divides by ε both sides of an equation.)

**Comment by safiire** [comment]
> In short, ε does not have an inverse.

Yes, your explanation convinces me. :)  That makes sense.
